trade-finance-manager-ui/server/routes/utilisation-reports/index.js

```diff
@@ -1,13 +1,20 @@
 const express = require('express');
-const { getUtilisationReports } = require('../../controllers/utilisation-reports');
-const { validateMongoId, validateUserTeam } = require('../../middleware');
-const { PDC_TEAM_IDS } = require('../../constants');
+const { getUtilisationReports, getUtilisationReportByBankId } = require('../../controllers/utilisation-reports');
 const { getReportDownload } = require('../../controllers/utilisation-reports/report-download');
+const { validateMongoId, validateUserTeam, validateBankId } = require('../../middleware');
+const { PDC_TEAM_IDS } = require('../../constants');
 
 const router = express.Router();
 
 router.get('/', validateUserTeam(Object.values(PDC_TEAM_IDS)), getUtilisationReports);
 
 router.get('/:_id/download', validateUserTeam(Object.values(PDC_TEAM_IDS)), validateMongoId, getReportDownload);
 
+router.get(
+  '/bank/:bankId',
```

Comment:
I wonder if we should make this endpoint specify the submission month (e.g. `/:submissionMonth/bank/:bankId` as its not clear what would be returned from this. (we can steal the `isoMonthValidation` from TFM API to validate it)

I know this is currently doing the same thing as the `/` endpoint by only returning results for the current submission month, but I think that was a mistake and will probably need to change when we add details of previous reports to the "Bank reports" page. What do you think?

Reply:
I agree that it will be a problem once we start including previous submission months. I will add the functionality at a basic level for this route and rebase on your previous submission months changes once they're in to finish up the logic (under the assumption that will be merged first)

Reply:
I think the previous reports work might still be a while. Is there any information you don't already have to be able to complete this without that new work?

Reply:
Is there a reason to not use the month and year as query parameters like `/bank/:bankId?sumissionMonth=&submissionYear=`?

Reply:
Yeah, good question. Some reasoning detailed here: https://github.com/UK-Export-Finance/dtfs2/pull/2366#discussion_r1419183543

Reply:
Reasoning sounds good to me - for my own sake, an example url would then be `/utilisation-reports/2023-12/bank/123`

Reply:
I have added in this functionality now